lib/core/src/Cardano/Wallet.hs

```diff
@@ -1420,7 +1420,12 @@ estimateFeeForDelegation ctx wid = db & \DBLayer{..} -> do
     let action = if isKeyReg then Join pid else RegisterKeyAndJoin pid
     let selectCoins = selectCoinsForDelegationFromUTxO @_ @t @k
             ctx utxo txp minUtxo action
-    estimateFeeForCoinSelection $ Fee . feeBalance <$> selectCoins
+
+    dep <- fmap stakeKeyDeposit $
+        withExceptT ErrSelectForDelegationNoSuchWallet
+            $ readWalletProtocolParameters @ctx @s @k ctx wid
```

Comment:
Note: this needs to be fed into the `initDelegationSelection` function to correctly initialize a coin selection with appropriate deposits (or reclaim). The only real difference with _before_ is that the deposit amount now comes directly from the "stakeKeyDeposit" protocol parameter, and not from the fee policy. Logic-wise, the rest should not change.

Reply:
It is:

```hs
    let selectCoins = selectCoinsForDelegationFromUTxO @_ @t @k
            ctx utxo txp minUtxo dep action
```